canvasapi/canvas_object.py

```diff
@@ -60,6 +62,16 @@ def set_attributes(self, attributes):
         """
         for attribute, value in attributes.items():
             self.__setattr__(attribute, value)
+            if attribute == "content-type":
+                self.__setattr__("content_type", value)
```

Comment:
On a cursory glance, I couldn't see any other response attributes that had hyphens. There are some other endpoints where camelCase is used (like the [result](https://canvas.instructure.com/doc/api/result.html) and [score](https://canvas.instructure.com/doc/api/score.html) endpoints), but those are only to report to IMS and LTI tools.

I think the question for me is which is more Pythonic - changing all the returned properties into class attributes or keeping the object as close to the returned structure (ie, keeping a hyphenated name and requiring the `getattr` method) from Canvas so the library matches the docs?

Reply:
I was also thinking of making it more generic, like if the value contained a hyphen to set it as a value with an underscore. I think I'm fine for making this into a future issue. 

It really _might_ be easiest to change this class to store all of the variables in a dictionary. Then the `__getattribute__` and `__getattr__` can be used to return from this dictionary by default. Then we can store both forms of the value with and without underscores. I'm fine with that as a separate issue and I don't think it'd be hard to change.

Reply:
The original PR proposed the general replace-all-hyphens-with-underscores. I also favour that solution. Canvas might add new attributes that have hyphens instead of underscores.

I also think it makes the most sense to keep the hyphenated version and provide the underscored version for convenience, to match the documentation as mentioned above. But it's important to get it right so that they sync, in case they're changed (although it doesn't make sense to change this particular attribute).